src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -262,8 +253,13 @@ private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage reques
                         byte[] buffer = await request.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(true);
                         cancellationToken.ThrowIfCancellationRequested();
 
-                        Task<JSObject> fetchPromise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, buffer);
-                        fetchResponse = await BrowserHttpInterop.CancelationHelper(fetchPromise, cancellationToken).ConfigureAwait(true);
+                        Memory<byte> bufferMemory = buffer.AsMemory();
+                        using (MemoryHandle pinBuffer = bufferMemory.Pin())
+                        {
+                            Task<JSObject> fetchPromise = BrowserHttpInterop.FetchBytes(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, pinBuffer, buffer.Length);
+                            // await must be here, otherwise bufferMemory.Pin() would release too early
+                            fetchResponse = await BrowserHttpInterop.CancelationHelper(fetchPromise, cancellationToken).ConfigureAwait(true);
```

Comment:
we should rename that thing to CancellationHelper I think

Reply:
typo? the diff shows this one wasn't renamed but the one above was

Reply:
Thanks, the same typo was on multiple places and I renamed only part of them.